core/validation/failures/user.py

```diff
@@ -266,140 +114,16 @@ class UnauthorisedSubmissionFailure(PreTransFormationFailure):
-
-    # group cells by sheet, section and desc
-    error_messages = group_validation_messages(error_messages)
-    error_messages.sort()
-    validation_errors = [
-        {"sheet": sheet, "section": section, "cell_index": cell_index, "description": message, "error_type": error_type}
-        for sheet, section, cell_index, message, error_type in error_messages
-    ]
-
-    return {"validation_errors": validation_errors}
-
-
-def group_validation_messages(
-    validation_messages: list[tuple[str, str, str, str, str]]
-) -> list[tuple[str, str, str, str, str]]:
-    """Groups validation messages by concatenating the cell indexes together on identical sheet, section and description
-
-    :param validation_messages: a list of tuples representing validation messages: sheet, section, description, cell
-    :return: grouped validation messages
-    """
-    grouped_dict = {}
-    for sheet, section, cell, desc, error_type in validation_messages:
-        key = (sheet, section, desc, error_type)  # use sheet, section, description and error_type as the key
-        value = cell  # use the cell index as the value
-        if key in grouped_dict:
-            grouped_dict[key].append(value)  # collect cells to concatenate
-        else:
-            grouped_dict[key] = [value]
-
-    grouped_messages = [
-        (sheet, section, ", ".join(sorted(cells)), desc, error_type)
-        for (sheet, section, desc, error_type), cells in grouped_dict.items()
-    ]
-
-    return grouped_messages
-
-
-def construct_cell_index(table: str, column: str, row_index: int) -> str:
-    """Constructs the index of an error from the column and row it occurred in.
-
-    :param table: the internal table name where the error occurred
-    :param column: the internal column name where the error occurred
-    :param row_index: a row index where the error occurred
-    :return: indexes tuple of constructed letter and number indexes
-    """
-    column_letter = TABLE_AND_COLUMN_TO_ORIGINAL_COLUMN_LETTER[table][column]
-    return column_letter.format(i=row_index)
-
 
-def remove_errors_already_caught_by_null_failure(
-    errors: list[tuple[str, str, str, str, str]]
-) -> list[tuple[str, str, str, str, str]]:
-    """
-    Removes errors from the list that have already been caught by null failures based on their sheet, section, and
-    cell index, and keeps only those present in null_failures. Additionally, includes all null_failures and errors
-    not present in null_failures or any part of the cell index is not already captured by a null failure.
-
-    :param errors: List of error tuples (sheet, section, cell_index, message, error_type).
-    :return: Filtered list of errors, including all null_failures and errors not present in null_failures or any part
-    of the cell index is not already captured by a null failure.
-    """
+@dataclass
+class TownsFundRoundFourValidationFailure(UserValidationFailure):
+    """Generic Towns Fund Round 4 Validation Failure."""
```

Comment:
I don't think this should be TF specific - I expect new funds will also want to be able to instantiate a generic failure with the same signature - so lets just make this `GenericUserFailure` or similar I think?

Reply:
Agreed, we have two types of "Generic" failures with different signatures one with all the message return values instantiated and this new one with column and row_index. I dont' think there is some neet way to overload or combine them though so I think both implementations must stay and we use different names for each